rest_framework/fields.py

```diff
@@ -1270,6 +1274,70 @@ def to_internal_value(self, data):
         return data
 
 
+class RecursiveField(Field):
+    """
+    A field that gets its representation from its parent.
+
+    This method could be used to serialize a tree structure, a linked list, or
+    even a directed acyclic graph. As with all recursive things, it is
+    important to keep the base case in mind. In the case of the tree serializer
+    example below, the base case is a node with an empty list of children. In
+    the case of the list serializer below, the base case is when `next==None`.
+    Above all, beware of cyclical references.
+
+    Examples:
+
+    class TreeSerializer(self):
+        children = ListField(child=RecursiveField())
+
+    class ListSerializer(self):
+        next = RecursiveField(allow_null=True)
+    """
+
+    def __init__(self, to='self', to_module=None, **kwargs):
+        self.to = to
+        self.to_module = to_module
+        field_kwargs = dict(
+            (key, kwargs[key])
+            for key in kwargs
+            if key in inspect.getargspec(Field.__init__)
+        )
```

Comment:
We can't pass the keyword arguments directly within `__init__`, because the field is being initialized within the serializer definition, so at this point in time the proxied serializer class may not even exist yet. 

In order for validation to work correctly, `Serializer` requires certain attributes to be available on the field itself (`read_only`, `allow_null`, `field_name`, `source`, etc..) -- either defined directly on the recursive field or forwarded through the recursive field to the proxied field. This is in addition to the list of methods I mentioned. The list of items was long enough that I did not want to forward them one at a time

I decided that calling the base Field initializer with the same arguments would ensure that the fields were present on `RecursiveField` without needing to forward them one-by-one. I did the same thing with `bind`, since `bind` is the other method which will potentially set attributes on the object itself.